puppet/manifests/overcloud_controller.pp

```diff
@@ -412,8 +412,12 @@
     include ::neutron::agents::dhcp
     include ::neutron::agents::metadata
 
+    $dnsmasq_options = hiera('neutron_dnsmasq_options', '')
+
+    # We need to create the dnsmasq-neutron.conf file regardless of
+    # whether there are configured options or the dhcp agent will fail.
     file { '/etc/neutron/dnsmasq-neutron.conf':
-      content => hiera('neutron_dnsmasq_options'),
+      content => $dnsmasq_options,
```

Comment:
you can just add the`, ' ' ` at the end of the hiera function like you did in the pacemaker version below